src/Uno.UWP/Devices/Midi/MidiInPort.iOSmacOS.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Runtime.InteropServices.WindowsRuntime;
+using System.Text;
+using System.Threading.Tasks;
+using CoreMidi;
+using Uno.Devices.Enumeration.Internal;
+using Uno.Devices.Enumeration.Internal.Providers.Midi;
+using Windows.Storage.Streams;
+
+namespace Windows.Devices.Midi
+{
+    public partial class MidiInPort
+    {
+		private MidiEndpoint _endpoint;
+		private MidiClient _client;
+		private MidiPort _port;
+
+		private MidiInPort(string deviceId, MidiEndpoint endpoint)
+		{
+			DeviceId = deviceId;
+			_endpoint = endpoint;
+			_client = new MidiClient(Guid.NewGuid().ToString());
+		}
+
+		internal void Open()
+		{
+			_port = _client.CreateInputPort(_endpoint.EndpointName);
+			_port.MessageReceived += NativePortMessageReceived;
+		}
+
+		private void NativePortMessageReceived(object sender, MidiPacketsEventArgs e)
+		{
+			foreach (var packet in e.Packets)
+			{
+				var bytes = new byte[packet.Length];
+				Marshal.Copy(packet.Bytes, bytes, 0, packet.Length);
+				OnMessageReceived(bytes, TimeSpan.FromMilliseconds(packet.TimeStamp));
+			}
+		}
+
+		public void Dispose()
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Either implement 'IDisposable.Dispose', or totally rename this method to prevent confusion.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5305560)